toplevel/coqtop.ml

```diff
@@ -207,9 +207,14 @@ let coqtop_parse_extra extras =
   let async_opts, extras = Stmargs.parse_args ~init:Stm.AsyncOpts.default_opts extras in
   ({ run_mode; color_mode}, async_opts), extras
 
+let fix_windows_dirsep s =
+  if Sys.win32 then Str.(global_replace (regexp_string "\\") "/" s)
+  else s
+
```

Comment:
Umm, the below code is already using `Filename.concat` which should do the right thing on Win32, so this seems wrong. Looking at the original bug, the problem seems to be in the Makefiles that mess with quoting, so it seems to me that the problem should be handled there.

In fact, this whole operation `get_native_name` seems to be ultra-costly, as it loads the whole summary, etc... the name of the native file can be easily computed without calling Coq [this is what we do in Dune] and indeed it seems like a better approach to me.

Reply:
> In fact, this whole operation `get_native_name` seems to be ultra-costly, as it loads the whole summary, etc... the name of the native file can be easily computed without calling Coq [this is what we do in Dune] and indeed it seems like a better approach to me.

Are you saying that you coded, in dune, the whole `-R` and `-Q` resolution strategy?

Reply:
No, dune already knows about these flags natively, so the operation of producing the name for native is quite easy:

https://github.com/ocaml/dune/blob/86522e99241cea205cc3202e419e6a6e003cc3e8/src/dune_rules/coq_module.ml#L43-L47

Indeed having a bit more information about library resolution would be very interesting as it would solve the current problem of having to generate all source files before calling coqdep, but that's offtopic here.

As far as I can see the bug is a bug in the Makefiles, that are quoting incorrectly somewhere, so IMO it should be solved there, not in Coq which is doing the right thing. [If filename.concat is buggy upstream that's a different story, but then we should provide our own, not hack ad-hoc here]